### PR TITLE
decouple scda

### PR DESCRIPTION
**Pull request by kartikeyakirar**
fixes https://github.com/insightsengineering/teal.data/issues/145

related PRs 
https://github.com/insightsengineering/scda/pull/131
https://github.com/insightsengineering/scda.2022/pull/129

### CODE REVIEW

File: R/CallablePythonCode.R [hunk 1]
```diff
@@ -176,15 +176,19 @@ PythonCodeClass <- R6::R6Class( # nolint
...
-#'
-#' x <- scda_cdisc_dataset_connector("ADSL", "adsl")
+#' ADSL <- teal.data::rADSL
+#' pull_adsl <- function(ADSL) ADSL <- teal.data::rADSL
+#' x <- dataset_connector(
+#'   dataname = "ADSL",
+#'   pull_callable = callable_function(fun = pull_adsl),
+#'   keys = get_cdisc_keys("ADSL"),
+#'   label = "Example label"
+#' )
```
Comment (gogonzo) [inline R/CallablePythonCode.R:191]: I think, you can leverage `example_cdisc_data` and `fun_dataset_connector` here and reuse in this 

```suggestion
#' random_data_connector <- function(dataname) {
#'   fun_dataset_connector(
#'     dataname = dataname,
#'     fun = example_cdisc_data,
#'     fun_args = list(dataname = dataname),
#'   )
#' }
#' ADSL <- random_data_connector(dataname = "ADSL")
```

File: R/CallablePythonCode.R [hunk 2]
```diff
@@ -196,9 +200,26 @@ PythonCodeClass <- R6::R6Class( # nolint
...
+#'   pull_callable = callable_function(fun = pull_adlb) %>% # nolint
+#'     set_args(list(ADLB = as.name("ADLB"))),
+#'   keys = get_cdisc_keys("ADLB"),
+#'   label = "ADLB connector"
+#' )
 #' tc <- cdisc_data(
-#'   scda_cdisc_dataset_connector("ADSL", "adsl"),
-#'   scda_cdisc_dataset_connector("ADLB", "adlb")
+#'   adsl_connector,
+#'   adlb_connector
```
Comment (gogonzo) [inline R/CallablePythonCode.R:222]:
```suggestion
#'   random_data_connector(dataname = "ADSL"),
#'   random_data_connector(dataname = "ADSL")
```

File: vignettes/using-delayed-data-advanced.Rmd [hunk 3]
```diff
@@ -408,26 +412,59 @@ if (interactive()) {
...
+)
+
+pull_adtte <- function(ADTTE) ADTTE <- teal.data::rADTTE
+adtte <- dataset_connector(
+  dataname = "ADTTE",
+  pull_callable = callable_function(fun = pull_adtte) %>%
+    set_args(list(ADTTE = as.name("ADTTE"))),
+  keys = get_cdisc_keys("ADTTE"),
+  label = "ADTTE connector"
+)
```
Comment (gogonzo) [inline vignettes/using-delayed-data-advanced.Rmd:464]: There is a `cdisc_dataset_connector` which can shorten this code. `pull_adsl` also can be simplified by the `example_cdisc_data`.

```suggestion
adsl <- cdisc_dataset_connector(
  dataname = "ADSL",
  pull_callable = callable_function(function() example_cdisc_data("ADSL")),
  label = "ADSL connector"
)

adae <- cdisc_dataset_connector(
  dataname = "ADAE",
  pull_callable = callable_function(function() example_cdisc_data("ADAE")),
  label = "ADAE connector"
)

advs <- cdisc_dataset_connector(
  dataname = "ADVS",
  pull_callable = callable_function(function() example_cdisc_data("ADVS")),
  label = "ADVS connector"
)

adtte <- dataset_connector(
  dataname = "ADTTE",
  pull_callable = callable_function(function() example_cdisc_data("ADTTE")),
  label = "ADTTE connector"
)
```

File: vignettes/using-delayed-data-advanced.Rmd [hunk 4]
```diff
@@ -566,8 +596,28 @@ as.character(adsl_code) == as.character(adtte_code)
...
+
+pull_adrs <- function(ADRS, n) ADRS <- head(teal.data::rADRS, n)
+adrs <- dataset_connector(
+  dataname = "ADRS",
+  pull_callable = callable_function(fun = pull_adrs) %>%
+    set_args(list(ADRS = as.name("ADRS"))),
+  keys = get_cdisc_keys("ADRS"),
+  label = "ADRS connector"
+)
```
Comment (gogonzo) [inline vignettes/using-delayed-data-advanced.Rmd:619]: please use `example_cdisc_data` and `cdisc_dataset_connector`

**Review by gogonzo — COMMENTED** [review]
Please make an advantage of functions we already have to shorten vignettes

File: vignettes/using-delayed-data-advanced.Rmd [hunk 5]
```diff
@@ -408,26 +412,51 @@ if (interactive()) {
...
+  keys = get_cdisc_keys("ADVS"),
+  label = "ADVS connector"
+)
+
+adtte <- cdisc_dataset_connector(
+  dataname = "ADTTE",
+  pull_callable = callable_function(function() example_cdisc_data("ADTTE")),
+  keys = get_cdisc_keys("ADTTE"),
+  label = "ADTTE connector"
+)
```
Comment (gogonzo) [inline vignettes/using-delayed-data-advanced.Rmd:456]: let's shorten this code

```suggestion
example_cdisc_dataset_connector <- function(dataname) {
  example_cdisc_dataset_connector(
    dataname = dataname,
    pull_callable = callable_function(function() example_cdisc_data(dataname)),
    keys = get_cdisc_keys(dataname),
    label = sprintf("%s connector")
  )
}
adsl <- example_cdisc_dataset_connector("ADSL")
adae <- example_cdisc_dataset_connector("ADAE")
advs <- example_cdisc_dataset_connector("ADVS")
adtte <- example_cdisc_dataset_connector("ADTTE")
```

File: vignettes/using-delayed-data-advanced.Rmd [hunk 6]
```diff
@@ -408,26 +412,36 @@ if (interactive()) {
...
 One can combine multiple delayed data objects using `[teal|cdisc]_data` functions and include them in a shiny application. `TealData` gathers all objects and sets combined UI with shiny inputs from all objects. In the code chunk below we specified four dataset connectors, but it works with arbitrary many objects of any combination of the classes `TealDataset`, `TealDatasetConnector` and / or `TealDataConnector`.
 
 ```{r}
-adsl <- scda_cdisc_dataset_connector("ADSL", "adsl")
-adae <- scda_cdisc_dataset_connector("ADAE", "adae")
-advs <- scda_cdisc_dataset_connector("ADVS", "advs")
-adtte <- scda_cdisc_dataset_connector("ADTTE", "adtte")
+# nolint start
+example_cdisc_dataset_connector <- function(dataname) {
+  example_cdisc_dataset_connector(
```
Comment (gogonzo) [inline vignettes/using-delayed-data-advanced.Rmd:431]:
```suggestion
  cdisc_dataset_connector(
```

File: vignettes/using-delayed-data-basic.Rmd [hunk 7]
```diff
@@ -82,7 +103,17 @@ Notes:
...
 
 ```{r}
-adsl <- scda_cdisc_dataset_connector(dataname = "ADSL", "adsl")
+# nolint start
+library(teal.data)
+library(magrittr)
+
+adsl <- cdisc_dataset_connector(
+  dataname = "ADSL",
+  pull_callable = callable_function(function() example_cdisc_data("ADSL")),
```
Comment (gogonzo) [inline vignettes/using-delayed-data-basic.Rmd:112]: This connector is badly specified. 
```
adsl$get_code()
# "ADSL <- (function() example_cdisc_data(\"ADSL\"))()"
```

Connector should be specified with some function and set_args
```r
adsl <- cdisc_dataset_connector(
  dataname = "ADSL",
  pull_callable = callable_function(teal.data::example_cdisc_data)$set_args(list(dataname = "ADSL")),
  keys = get_cdisc_keys("ADSL"),
  label = "ADSL connector"
)
adsl$get_code()
"ADSL <- teal.data::example_cdisc_data(dataname = \"ADSL\")"
```

Comment (kartikeyakirar) [reply vignettes/using-delayed-data-basic.Rmd]: Yes , this is updated. 
some places I have used with magrittr  pipe operator
```
callable_function(example_cdisc_data) %>%
  set_args(list(dataname = "<dataname>").
```

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**